graph/patterns/longRunningOperations.md

```diff
@@ -0,0 +1,243 @@
+
+- For RELO pattern you should return the Location header that indicates the location of the resource.
+  - The API response says the targeted resource is being created by returning a 201 status code and the resource URI is provided in the Location header, but the response indicates that the request is not completed by including "Provisioning" status.
+
+- For LRO pattern you should return the Location header that indicates the location of a new stepwise operation resource.
+  - The API response says the operation resource is being created at the URL provided in the Location header and indicates that the request is not completed by including a 202 status code.
+  - Microsoft Graph doesn’t allow tenant wide operation resources therefore stepwise operations is often modeled as a navigation property on the target resource.
+
+## When to use this pattern
+
+Any API call that is expected to take longer than 1 second in the 99th percentile should use the long running operations pattern.
+
+How do you select which flavor of LRO pattern to use? An API designer can follow these
+heuristics:
+
+1.  If a service can create a resource with a minimal latency and continue
+    updating its status according to the well-defined and stable state
+    transition model until completion, then the RELO model is the best choice.
+
+2.  Otherwise, a service should follow the Stepwise Operation pattern. 
+ 
+
+## Issues and considerations
+
+- One or more API consumers MUST be able to monitor and operate on the same resource at the same time.
+
+- The state of the system SHOULD always be discoverable and testable. Clients
+    SHOULD be able to determine the system state even if the operation tracking
+    resource is no longer active. Clients MAY issue a GET on some resource to
+    determine the state of a long running operation.
+
+-  The long running operations pattern SHOULD work for clients looking to "fire and forget"
+    and for clients looking to actively monitor and act upon results.
+
+-  The long running operations pattern might be supplemented by the [change notification
+    pattern](change-notification.md).
+
+-  Cancellation of a long running operation does not explicitly mean a rollback. On a per API-defined case, it
+    might mean a rollback, or compensation, or completion, or partial completion,
+    etc. Following a canceled operation, the API should return a consistent state that allows
+    continued service.
+
+-  A recommended minimum retention time for a stepwise operation is 24 hours.
+    Operations SHOULD transition to "tombstone" for an additional period of time
+    prior to being purged from the system.
+    
+-  Services that provides a new operation resource MUST support GET semantics on the operation.
+-  Services that returns a new operation MUST always return an LRO (even if the LRO is created in the completed state) that way API consumers don't have to deal with two different shapes of response.
+
+
+
+## Examples
```

Comment:
@darrelmiller while attempting to add hybrid model we identified that because the response headers contain both tatget resource URL and operation URL there is a need for the 'status' property on both resources. Otherwise a client can't determine if the target resource isn't ready by using GET on target URL. duplication of status property is an antipattern and would require additional sync work. Therefore we (Peter, Dan, Gareth, Olga) collectively decided not to include hybrid model in the pattern